instance/factories.py

```diff
@@ -101,6 +101,7 @@ def instance_factory(**kwargs):
     assert is_valid_domain_name(kwargs.get('sub_domain'))
 
     # Create instance
+    # n = OpenEdXInstance.objects.create(instance_purpose="3")
```

Comment:
@arbrandes I'm trying to make it possible to instantiate a new OpenEDX instance with a specific instance_purpose but somehow doing this coughs out a TypeError claiming `OpenEdXInstance() got an unexpected keyword argument 'instance_purpose'`. I have defined a property and setter in the Instance class and I expect this value to be available in the sub classes but thats not the case. Note, this only happens when I'm creating a new instance, I can get, view, and set the instance_purpose of an existing OpenEDXInstance but the instance_purpose property doesn't exist yet at creation because it's created when we create a new InstanceReference for this oedxinstance at the Instance.ref() property. Question: How do I propagate this value (instance_purpose) from the OpenEDXInstance, through the ref property in Instance to create an InstanceReference with the desired instance_purpose value at creation time?